src/plugins/irc-events/link.js

```diff
@@ -163,6 +163,13 @@ function fetch(uri, cb) {
 				// if not image, limit download to 50kb, since we need only meta tags
 				// twitter.com sends opengraph meta tags within ~20kb of data for individual tweets
 				limit = 1024 * 50;
+			} else {
+				// response is an image
+				// if Content-Length header reports a size exceeding the prefetch limit, abort fetch
+				const contentLength = parseInt(res.headers["content-length"], 10) || NaN;
```

Comment:
parseInt returns NaN on it's own, I would change it to 0 though, then you don't need to the AND check below.

Reply:
Testing in Chrome's dev console with a simple literal object, running parseInt on a key that didn't exist set the test var to `undefined`. But using 0 instead is sensible. I went with NaN and the `&&` in case a negative limit (i.e. no limit) is/will be possible.